packages/commands/src/components/command-menu.js

```diff
@@ -187,45 +183,34 @@ function CommandInput( { isOpen, search, setSearch } ) {
  * @ignore
  */
 export function CommandMenu() {
-	const { registerShortcut } = useDispatch( keyboardShortcutsStore );
 	const [ search, setSearch ] = useState( '' );
-	const isOpen = useSelect(
-		( select ) => select( commandsStore ).isOpen(),
-		[]
-	);
+	const isOpenRef = useRef( false );
+	const isOpen = useSelect( ( select ) => {
+		return ( isOpenRef.current = select( commandsStore ).isOpen() );
+	}, [] );
 	const { open, close } = useDispatch( commandsStore );
 	const [ loaders, setLoaders ] = useState( {} );
 
 	useEffect( () => {
-		registerShortcut( {
-			name: 'core/commands',
-			category: 'global',
-			description: __( 'Open the command palette.' ),
-			keyCombination: {
-				modifier: 'primary',
-				character: 'k',
-			},
-		} );
-	}, [ registerShortcut ] );
+		const handleUserKeyPress = ( e ) => {
+			if ( e.metaKey && e.key === 'k' ) {
```

Comment:
Check using `event.metaKey` does not work on Windows. We also need to check `event.ctrlKey`. I'm thinking of using the `isKeyboardEvent` function from the `@wordpress/keycodes` package to check as follows.

```javascript
if ( isKeyboardEvent.primary( e, 'k' ) ) {
}
```

Reply:
Good call, I'll revert to the useShortcut hooks and normally if that works across the board we will benefit from that library as well.